cmd/pj-rehearse/main.go

```diff
@@ -1,23 +1,278 @@
 package main
 
 import (
+	"encoding/json"
 	"flag"
 	"fmt"
-	"log"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	// TODO: Solve this properly
+	"github.com/getlantern/deepcopy"
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	pjclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	pj "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/pjutil"
+
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/openshift/ci-operator-prowgen/pkg/diffs"
 )
 
+func getPrNumber(jobSpec *pjapi.ProwJobSpec) int {
```

Comment:
Why is this useful as a function?

Reply:
It's more readable at callsite than the ugly expression it wraps.